includes/plugins/wc-memberships/class-memberships.php

```diff
@@ -774,15 +773,18 @@ public static function render_js() {
 				ras.on( 'reader', function( ev ) {
 					if ( ev.detail.authenticated && ! window?.newspackReaderActivation?.getPendingCheckout() ) {
 						if ( ras.overlays.get().length ) {
+							// When an overlay is added or removed,
+							// check if there are none – this means an overlay was removed and there are none left.
+							// In this case, reload the window.
 							ras.on( 'overlay', function( ev ) {
-								if ( ! ev.detail.overlays.length && ! window?.newspackReaderActivation?.openNewslettersSignupModal ) {
```

Comment:
@chickenn00dle – please advise if this condition was needed. It appears to be a constant condition (`openNewslettersSignupModal` is always defined), so the reload would never happen. I can't see the reason for this condition, and there's no comment to shed light on it.